### ISSUE TITLE
Probem on psds_score_scenario2

### ISSUE THREAD

**Issue by Aiksier** [DCASE-REPO/DESED_task#51]
Dear organizer:
    When I using weak label to compute psds score scenario2, I got some problem, I don't know how to fix it. Thanks for your time and kindly help~
   (1) Transfer learning from PANNs to train an audio tagging system，using weak label （torch.ones_like(ground_truth strong_labels) * weak_preds） as the predict result to compute psds_score_scenario2:
DATALOADER:0 TEST RESULTS
{'hp_metric': 0.7078940868377686,
 'test/student/event_f1_macro': 0.21531902253627777,
 'test/student/intersection_f1_macro': 0.8512769937515259,
 'test/student/psds_score_scenario1': 0.04927980527281761,
 'test/student/psds_score_scenario2': 0.7078940868377686,
 'test/teacher/event_f1_macro': 0.21711485087871552,
 'test/teacher/intersection_f1_macro': 0.8789225220680237,
 'test/teacher/psds_score_scenario1': 0.04864690080285072,
 'test/teacher/psds_score_scenario2': 0.7067073583602905}

![image](https://user-images.githubusercontent.com/68141643/167819490-4b8dfaa1-8d5c-4c8a-b3c8-41401b832a1f.png)
(2) Using ground truth strong label in test set to get ground truth weak label, and compute the psds_score_scenario2 using ground truth weak label（torch.ones_like(ground_truth strong_labels) * ground_truth_weak_label）.
DATALOADER:0 TEST RESULTS
{'hp_metric': 0.7472647428512573,
 'test/student/event_f1_macro': 0.24901337921619415,
 'test/student/intersection_f1_macro': 0.980992317199707,
 'test/student/psds_score_scenario1': 0.04510604217648506,
 'test/student/psds_score_scenario2': 0.7472647428512573,
 'test/teacher/event_f1_macro': 0.24901337921619415,
 'test/teacher/intersection_f1_macro': 0.980992317199707,
 'test/teacher/psds_score_scenario1': 0.04510604217648506,
 'test/teacher/psds_score_scenario2': 0.7472647428512573}
![image](https://user-images.githubusercontent.com/68141643/167819586-0f26a559-e7d4-4293-83b2-334b55f27bb0.png)

**Comment by Aiksier** [comment]
Thanks for your reply. The weak lables are broadcast to a ones like strong predictions matrix to conpute psds scores.


best wishes~



---Original---
From: ***@***.***&gt;
Date: Mon, Jul 4, 2022 17:32 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DCASE-REPO/DESED_task] Probem on psds_score_scenario2 (Issue#51)




 
Hi,
 Very sorry about the late answer.
 I'm a bit confused, how do you compute PSDS when using weak labels ?

**Comment by turpaultn** [comment]
Closed due to inactivity
